be/src/pipeline/exec/exchange_sink_buffer.cpp

```diff
@@ -197,53 +197,91 @@ Status ExchangeSinkBuffer::_send_rpc(InstanceLoId id) {
-    _closure->cntl.set_timeout_ms(request.channel->_brpc_timeout_ms);                              \
-    _closure->addFailedHandler(                                                                    \
-            [&](const InstanceLoId& id, const std::string& err) { _failed(id, err); });            \
-    _closure->addSuccessHandler([&](const InstanceLoId& id, const bool& eos,                       \
-                                    const PTransmitDataResult& result) {                           \
-        Status s = Status(result.status());                                                        \
-        if (!s.ok()) {                                                                             \
-            _failed(id,                                                                            \
-                    fmt::format("exchange req success but status isn't ok: {}", s.to_string()));   \
-        } else if (eos) {                                                                          \
-            _ended(id);                                                                            \
-        } else {                                                                                   \
-            _send_rpc(id);                                                                         \
-        }                                                                                          \
-    });                                                                                            \
-    {                                                                                              \
-        SCOPED_SWITCH_THREAD_MEM_TRACKER_LIMITER(ExecEnv::GetInstance()->orphan_mem_tracker());    \
-        if (enable_http_send_block(*brpc_request)) {                                               \
-            RETURN_IF_ERROR(transmit_block_http(_context->get_runtime_state(), _closure,           \
-                                                *brpc_request, request.channel->_brpc_dest_addr)); \
-        } else {                                                                                   \
-            transmit_block(*request.channel->_brpc_stub, _closure, *brpc_request);                 \
-        }                                                                                          \
-    }                                                                                              \
-    if (request.BLOCK) {                                                                           \
-        brpc_request->release_block();                                                             \
-    }                                                                                              \
-    QUEUE.pop();
-
     if (!q.empty()) {
         // If we have data to shuffle which is not broadcasted
-        DO_RPC(q, block.get(), nullptr)
```

Comment:
if you remove the DO_RPC, maybe has the similar code.

Reply:
I did this because some bugs occurs and the stack reported by ASAN is not accurate due to this macro. So I just remove this macro to help me find out what's wrong here. I will revert this PR after this.